### PR TITLE
Add new booking models & migrations

### PR DESCRIPTION
**Pull request by chx2**
Added updated models & migrations for the package.

### CODE REVIEW

File: database/migrations/2020_05_04_150000_create_bookings_table.php [hunk 1]
```diff
@@ -5,24 +5,33 @@
...
-            $table->foreignIdFor(app('tax'))->nullable(); // Tax rate used on this booking
-            $table->foreignIdFor(app('fee'))->nullable(); // Processing fee used on this booking
+            $table->tinyInteger('slot_number');
+            $table->tinyInteger('total_participants');
+            $table->integer('total_amount');
+            $table->integer('amount');
+            $table->integer('total_taxes');
+            $table->integer('total_fees');
+            // Todo: Add morph relation foreign IDs
+//            $table->foreignIdFor(app('variation'))->nullable();
```
Comment (chx2) [inline database/migrations/2020_05_04_150000_create_bookings_table.php]: Commented these out for now for passing tests. WIll need to add references to morphs in the factory and tests eventually.

File: src/Models/Booking.php [hunk 2]
```diff
@@ -116,67 +62,34 @@ public function scopeWeekComparison($query)
...
-    {
-        $date = strtoupper($this->slot->getCarbonStartAt()->setTimezone($this->slot->room->location->php_tz)->format('D\, M j'));
-        $arrival = $this->slot->getCarbonStartAt()->setTimezone($this->slot->room->location->php_tz)->subMinutes(15)->format('g:i A');
-        $start = $this->slot->getCarbonStartAt()->setTimezone($this->slot->room->location->php_tz)->format('g:i A');
-
-        return $date.' ⬩ ARRIVE BY '.$arrival.' ⬩ STARTS AT '.$start;
-    }
-
-    public function getSlotStartAttribute()
+    // Todo: Determine Morph Relation Type
```
Comment (chx2) [inline src/Models/Booking.php:65]: What morph types should be used?

File: src/Models/Booking.php [hunk 3]
```diff
@@ -5,85 +5,31 @@
...
-    }
-
-    /**
-     * Generate amount, total_taxes and total_fees.
-     *
-     * !!! This funcionaly is uswed mainly by seeder !!!
-     *
-     * @return void
-     */
-    public function generatePricing()
```
Comment (chx2) [inline src/Models/Booking.php:47]: I removed a significant portion of the existing methods. Not sure how many are still relevant based on the updated model structure.

**Comment by drewroberts** [comment]
I'm going to merge this into the other related PR

**Review by drewroberts — APPROVED** [review]
Thank you! 🚉

**Pull request merged**